uint64 id
int8 type
SensorSpec[] sensorSpec
CommSpec[] commSpec
AppSpec[] appSpec
